pynisher/limit_function_call.py

```diff
@@ -284,10 +284,18 @@ def read_connection():
 
                     # recover stdout and stderr if requested
                     if self.capture_output:
-                        with open(os.path.join(tmp_dir.name, 'std.out'), 'r') as fh:
-                            self2.stdout = fh.read()
-                        with open(os.path.join(tmp_dir.name, 'std.err'), 'r') as fh:
-                            self2.stderr = fh.read()
+                        out_file = os.path.join(tmp_dir.name, 'std.out')
```

Comment:
Could you please use `try/except` here, that might be a bit more robust?

Reply:
of course, that's better. Done!